### PR TITLE
Add cookieJar option

### LINKED ISSUE #918: Add support for providing custom cookie jar
1. Newman Version =   3.4.3
 2. OS details (type, version, and architecture):  =   Windows 10 x64.
 3. Are you using Newman as a library, or via the CLI?   =  CLI
 3. Did you encounter this recently, or has this bug always been there:   probably always there
 4. Expected behaviour:  =  work as Postman does
 5. Command / script used to run Newman:   =  
newman run ".\dirname\dirname\PostmanFiles\Proj.postman_collection.json" --environment ".\dirname\dirname\PostmanFiles\Proj%%20localhost.postman_environment.json" --folder apiFunctions

 6. Sample collection, and auxilliary files (minus the sensitive details): = none relevant
 7. Screenshots (if applicable): = N/A

Steps to reproduce the problem:

1. Run above CLI command.

### PR DESCRIPTION
**Pull request by rohit-gohri**
Closes #918

- [x] Add basic implementation
- [x] Add tests
- [x] Update docs

### CODE REVIEW

**Comment by rohit-gohri** [comment]
@codenirvana Done from my side

File: package.json [hunk 1]
```diff
@@ -57,6 +57,7 @@
     "pretty-ms": "7.0.0",
     "semver": "7.3.2",
     "serialised-error": "1.1.3",
+    "tough-cookie": "3.0.1",
```
Comment (rohit-gohri) [inline package.json:60]: There is a v4 available, but since postman-runtime was using 3.0.1 I decided to keep it same as that.

**Comment by rohit-gohri** [comment]
@codenirvana Did you have a look?

File: bin/newman.js [hunk 2]
```diff
@@ -65,6 +65,10 @@ program
         'Specify the Client SSL passphrase (optional, needed for passphrase protected keys).')
     .option('--ssl-extra-ca-certs <path>',
         'Specify additionally trusted CA certificates')
+    .option('--cookie-jar <path>',
+        'Specify custom cookie jar file path')
```
Comment (codenirvana) [inline bin/newman.js:69]: This must be under 120 char length, new-line not required.

Also, should we mention that this expects a ToughCookie serialized jar here?

File: lib/run/index.js [hunk 3]
```diff
@@ -173,7 +177,7 @@ module.exports = function (options, callback) {
             fileResolver: new SecureFS(options.workingDir, options.insecureFileRead),
             requester: {
                 useWhatWGUrlParser: true,
-                cookieJar: request.jar(),
+                cookieJar: cookieJar,
```
Comment (codenirvana) [inline lib/run/index.js:180]:
```suggestion
                cookieJar: options.cookieJar || request.jar(),
```

File: test/fixtures/run/spaces/simple-cookie-jar.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{
+  "version": "tough-cookie@3.0.1",
+
+
```
Comment (codenirvana) [inline test/fixtures/run/spaces/simple-cookie-jar.json:4]: Unnecessary newlines 😅

File: lib/run/index.js (hunk 3 above)
Comment (rohit-gohri) [reply lib/run/index.js:180]: I have done this already when declaring the variable above this

Comment (rohit-gohri) [reply lib/run/index.js:180]: And I need to do that because that variable is being used for export too below

**Review by codenirvana — APPROVED** [review]

**Comment by rohit-gohri** [comment]
@codenirvana Thanks!

**Pull request merged**